src/cargo/ops/cargo_add/dependency.rs

```diff
@@ -592,9 +607,19 @@ impl Dependency {
                     })
                     .unwrap_or_default();
                 features.extend(new_features.iter().map(|s| s.as_str()));
-                let features = toml_edit::value(features.into_iter().collect::<toml_edit::Value>());
-                table.set_dotted(false);
-                table.insert("features", features);
+                if let Some(inherit_features) = &self.inherited_features {
+                    inherit_features.iter().for_each(|f| {
+                        features.remove(f.as_str());
+                    });
+                }
+                if !features.is_empty() {
+                    let features =
+                        toml_edit::value(features.into_iter().collect::<toml_edit::Value>());
+                    table.set_dotted(false);
+                    table.insert("features", features);
+                } else {
+                    table.remove("features");
+                }
```

Comment:
Not quite convinced this is the right route to go.  `inherit_features` is just advisory before this change

Reply:
This is my bad, I accidentally added this. It should've been moved to my shelf